arrows/klv/klv_read_write.txx

```diff
@@ -344,6 +357,114 @@ klv_ber_oid_length( T value )
   return ( _int_bit_length( value ) + 6 ) / 7;
 }
 
+// ---------------------------------------------------------------------------
+template < class T, class Iterator >
+double
+klv_read_flint( double minimum, double maximum, Iterator& data, size_t length )
+{
+  KLV_ASSERT_INT( T );
+  KLV_ASSERT_UINT8_ITERATOR( data );
+
+  _check_range_length( minimum, maximum, length );
+
+  // Before read to avoid moving iterator
+  if( std::is_signed< T >::value && minimum != -maximum )
+  {
+    throw std::logic_error( "range must be symmetrical around zero" );
+  }
+
+  auto const int_value = klv_read_int< T >( data, length );
+  auto const float_value = static_cast< double >( int_value );
+
+  if( std::is_signed< T >::value )
+  {
+    // Special invalid / out-of-range value
+    if( int_value == _int_min< T >( length ) )
+    {
+      return std::numeric_limits< double >::quiet_NaN();
+    }
+
+    auto const scale = maximum / _int_max< T >( length );
+    return float_value * scale;
+  }
+  else
+  {
+    auto const scale = ( maximum - minimum ) / _int_max< T >( length );
+    return float_value * scale + minimum;
+  }
+}
+
+// ---------------------------------------------------------------------------
+template < class T, class Iterator >
+void
+klv_write_flint( double value, double minimum, double maximum,
+                 Iterator& data, size_t length )
+{
+  // Ensure types are compatible with our assumptions
+  KLV_ASSERT_INT( T );
+  KLV_ASSERT_UINT8_ITERATOR( data );
+
+  _check_range_length( minimum, maximum, length );
+
+  // Check before NaN
+  if( std::is_signed< T >::value && minimum != -maximum )
+  {
+    throw std::logic_error( "range must be symmetrical around zero" );
+  }
+
+  // Check for NaN
+  auto const min_int = _int_min< T >( length );
+  auto const max_int = _int_max< T >( length );
+  if( std::isnan( value ) )
+  {
+    klv_write_int( min_int, data, length );
+    return;
+  }
+
```

Comment:
```suggestion
  if( std::isnan( value ) )
  {
    klv_write_int( min_int, data, length );
    return;
  }

  auto const min_int = _int_min< T >( length );
  auto const max_int = _int_max< T >( length );
```

Reply:
But we need `min_int` inside the if statement. We could put `min_int` before and `max_int` after? I had it like that at one point , but for me keeping the related declarations together overrode the tiny performance ?gain?.

Reply:
Oh, right; my bad. Anyway, this was more a reaction to the locals being immediately after the comment, which felt a little odd. Revised suggestion coming (on different lines, so needs to be a new comment).